CONTRIBUTING.md

```diff
@@ -42,6 +31,34 @@ Additionally, JS files need to be run through [JSBeautify](https://github.com/ei
 
 Any patch should be tested in as many of our [supported browsers](LINK HERE) as possible. Obviously, access to all devices is rare, so just aim for the best coverage possible. At a minimum please test in all available desktop browsers.
 
+### Running Tests
+
+To run tests you need a [global install of Phantom JS](http://phantomjs.org/download.html).
+
+Once PhantomJS is installed, simply execute `npm test`.
+
+### Adding Tests
+
+Any patch for this repo should be accompanied by one or more automated tests to detect and avoid regressions introduced by future code changes.
+
+Currently we use PhantomJS for functional testing in a headless browser. To add a new test, edit `test/phantom.js` and append a new `Test` object to the `tests` array.
+
+A `Test` object is instantiated with the following parameters: `description, url, injectedJS, testBody`
+
+- `description` (*String*) - Description of test success. (eg: "Clicking delete removes item.")
+- `url` (*String*) - The URL that the browser should open before the test is run.
+- `injectedJS` (*String*) - Defines a path to JS that should be injected into the page before the test is run. This could be mocked data, a sequence of interactions common to multiple tests, or something else your test needs to run.
+- `testBody` (*Function*) - A function containing your test code. Most likely this function will contain one or more calls to `page.evaluate`, which allows JS to be run in the context of the page being tested. This function requires a call to `this.onComplete()` with a boolean value representing the outcome of the test (eg: `this.onComplete(false)` for a failure).
+
+#### Targeting Elements
+
+The preferred way to target an element is to give it a `data-test-id` attribute with a globally unique identifier. This data attribute is only used in testing and should not be used in the primary application code!
```

Comment:
It's still there in prod. I realize this isn't totally ideal, but it doesn't really hurt anything...